extensions-core/google-extensions/src/main/java/org/apache/druid/storage/google/GoogleStorage.java

```diff
@@ -159,11 +167,23 @@ public void delete(final String bucket, final String path) throws IOException
    * @param bucket GCS bucket
    * @param paths  Iterable for absolute paths of objects to be deleted inside the bucket
    */
-  public void batchDelete(final String bucket, final Iterable<String> paths) throws IOException
+  public void batchDelete(final String bucket, final Iterable<String> paths)
   {
-    List<Boolean> statuses = storage.get().delete(Iterables.transform(paths, input -> BlobId.of(bucket, input)));
+    final List<Boolean> statuses = storage.get().delete(Iterables.transform(paths, input -> BlobId.of(bucket, input)));
```

Comment:
Same as above - could you please include javadocs for `batchDelete`  similar to what you'd have for the `delete` call? The underlying assumption is relying on undocumented behavior that a false indicates that a file is not found and any other error would result in an exception that needs to be handled explicitly.

Reply:
Added